src/components/ErrorBoundary/index.tsx

```diff
@@ -33,179 +31,78 @@ const CodeBlockWrapper = styled.div`
   color: ${({ theme }) => theme.deprecated_text1};
 `
 
-const LinkWrapper = styled.div`
-  color: ${({ theme }) => theme.deprecated_blue1};
+const Padding = styled.div`
   padding: 6px 24px;
 `
 
-const SomethingWentWrongWrapper = styled.div`
-  padding: 6px 24px;
-`
-
-type ErrorBoundaryState = {
-  error: Error | null
+const Fallback = ({ error }: { error: Error }) => {
```

Comment:
I think this component should include a reload button, using `reloadIfUpdateAvailable`-like code - but I think it should reload even if there isn't an update available.

Reply:
Yes, users will trigger reload either manually by pressing a button or CMD+R (or other combination). However, we will still run part of the code from that function in the background to actually update service workers in the background ahead of users reloading. 

The `update` function is async and actually downloads the actual code, so I expect this to take some time depending on the network connection.

Currently, reload happens after update is finished, which means that on a fast connection users don't see error page, on slower connection they have random reload after some time. 

Unfortunately, this means that it will be technically possible for users to press reload while service worker update is in progress, as there's no way for us to prevent a reload from happening. FWIW, this was happening already, so it's not a new behaviour, just something to keep in mind.